golang/cosmos/x/vstorage/keeper/keeper.go

```diff
@@ -366,17 +380,25 @@ func (k Keeper) SetStorage(ctx sdk.Context, entry agoric.KVEntry) {
 	store := ctx.KVStore(k.storeKey)
 	path := entry.Key()
 	encodedKey := types.PathToEncodedKey(path)
+	rawValue := store.Get(encodedKey)
 
 	if !entry.HasValue() {
 		if !k.HasChildren(ctx, path) {
 			// We have no children, can delete.
+			sizeDelta := float32(-len(encodedKey) - len(rawValue))
+			telemetry.IncrCounterWithLabels(MetricKeysSizeDelta, sizeDelta, NewMetricsLabels(k))
```

Comment:
I am surprised we're forced to use `float32` here since the max int value for these is `16777216`, which will definitely cause loss of precision after a few increments. But maybe internally they still deal in float64?

https://github.com/hashicorp/go-metrics/issues/164